app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,84 @@
+class ArticlesController < ApplicationController
+  before_action :set_article, only: [:show, :edit, :update, :destroy]
+  before_action :authenticate_user!, expect: [:index,:show]
+  # GET /articles
+  # GET /articles.json
+  def index
+    @articles = Article.all
+  end
+
+  # GET /articles/1
+  # GET /articles/1.json
+  def show
+  end
+
+  # GET /articles/new
+  def new
+    @article = Article.new
+  end
+
+  # GET /articles/1/edit
+  def edit
+  end
+
+  # POST /articles
+  # POST /articles.json
+  def create
+    @article = current_user.articles.build(article_params)
+    respond_to do |format|
+      if @article.save
+        if params[:images].present?
```

Comment:
@uzaif313 can you please extract this code to new private method i.e. save_image(image_param) ?